src/node_crypto.cc

```diff
@@ -255,13 +255,49 @@ void ThrowCryptoError(Environment* env,
                       unsigned long err,  // NOLINT(runtime/int)
                       const char* default_message = nullptr) {
   HandleScope scope(env->isolate());
+  Local<String> message;
+
   if (err != 0 || default_message == nullptr) {
     char errmsg[128] = { 0 };
     ERR_error_string_n(err, errmsg, sizeof(errmsg));
-    env->ThrowError(errmsg);
+    message = String::NewFromUtf8(env->isolate(), errmsg,
+                                  v8::NewStringType::kNormal)
+                                      .ToLocalChecked();
   } else {
-    env->ThrowError(default_message);
+    message = String::NewFromUtf8(env->isolate(), default_message,
+                                  v8::NewStringType::kNormal)
+                                      .ToLocalChecked();
+  }
+
+  Local<Value> exception_v = Exception::Error(message);
+  CHECK(!exception_v.IsEmpty());
+  Local<Object> exception = exception_v.As<Object>();
+  Local<Array> errorStack = Array::New(env->isolate());
+
+  ERR_STATE *es = ERR_get_state();
+  // Build the errorStack array to be added to openSSLErrorStack property.
+  for (unsigned int i = 0; es->bottom != es->top
+        && (es->err_flags[es->top] & ERR_FLAG_MARK) == 0; i++) {
+    unsigned long err_buf = es->err_buffer[es->top];  // NOLINT(runtime/int)
+    // Only add error string if there is valid err_buffer.
+    if (err_buf) {
+      char tmpStr[256] = { };
+      ERR_error_string_n(err_buf, tmpStr, sizeof(tmpStr));
+      errorStack->Set(i, String::NewFromUtf8(env->isolate(), tmpStr,
```

Comment:
And its ok that I get a compiler warning `warning: ignoring return value of function declared with 'warn_unused_result' attribute`?